.github/workflows/check_updates.java

```diff
@@ -0,0 +1,223 @@
+            }
+            String repository = tree.path("maven-repository").asText(MAVEN_CENTRAL);
+            String groupId = tree.get("group-id").asText();
+            String artifactId = tree.get("artifact-id").asText();
+            log.infof("Fetching latest version for %s:%s", groupId, artifactId);
+            ArrayNode versionsNode = tree.withArray("versions");
+            // Get Latest Version
+            String latestVersion = getLatestVersion(repository, groupId, artifactId, versionsNode);
+            if (latestVersion == null) {
+                log.warnf("%s:%s latest version was read previously (or could not find it). Skipping", groupId, artifactId);
+                log.info("---------------------------------------------------------------");
+                return;
+            }
+            // Compare if not already in descriptor
+            if (containsValue(versionsNode, latestVersion)) {
+                log.warnf("%s:%s version %s was read previously. Skipping", groupId, artifactId, latestVersion);
+                return;
+            } else {
+                versionsNode.insert(0, latestVersion);
+            }
+            // Write version
+            yamlMapper.writeValue(extensionYaml.toFile(), tree);
+            // Git commit
+            gitCommit(extensionYaml, "Add " + latestVersion + " to " + workingDirectory.resolve(extensionYaml).normalize());
+        } catch (IOException e) {
+            log.error("Error while processing extension", e);
+        }
+        log.info("---------------------------------------------------------------");
+    }
+
+    private boolean containsValue(ArrayNode versionsNode, String latestVersion) {
+        for (JsonNode node : versionsNode) {
+            if (latestVersion.equals(node.asText())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String getLatestVersion(String repository, String groupId, String artifactId, ArrayNode versionsRead) throws IOException {
+        URI metadataURL = URI.create(MessageFormat.format("{0}{1}/{2}/maven-metadata.xml",
+                                                          Objects.toString(repository, MAVEN_CENTRAL),
+                                                          groupId.replace('.', '/'),
+                                                          artifactId));
+        try (CloseableHttpClient httpClient = createHttpClient();
+             InputStream is = httpClient.execute(new HttpGet(metadataURL)).getEntity().getContent()) {
+            MetadataXpp3Reader metadataReader = new MetadataXpp3Reader();
+            Metadata metadata = metadataReader.read(is);
+            Versioning versioning = metadata.getVersioning();
+            String candidateVersion = versioning.getLatest();
+            if (!containsValue(versionsRead, candidateVersion)) {
+                return candidateVersion;
+            }
+            // Try the previous released version
```

Comment:
What kind of use-case is it for?

Reply:
When 2.0.4.Final is released, for example. The metadata XML always lists the greatest version as the latest

Reply:
Would it make sense to check all the versions?

Reply:
I thought about that and I think that checking all now makes sense because the `publish` script will always upload the first element in the list (the newest). I'll change it and give it a shot